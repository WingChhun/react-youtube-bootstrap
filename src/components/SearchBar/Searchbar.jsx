import React, {Component} from 'react';
import PropTypes from 'prop-types';

import YTSearch from 'youtube-api-search';

class Searchbar extends Component
{
    constructor(props)
    {
        super(props);

        this.state = {
            inputValue: ""
        }
    }

    handleChange = (event) => {

        const inputValue = event
            .target
            .value
            .trim();

        this.setState({inputValue});
    }

    onKeyPress = event => {

        const {API_KEY, getVideos} = this.props;
        const {inputValue} = this.state;

        if (event.key === "Enter") {

            //*Call search API
            console.log("User hit onKeypress enter!");

            YTSearch({
                key: API_KEY,
                term: inputValue
            }, (data) => {

                getVideos(data);

            })
        }
    }

    render()
    {

        const {inputValue} = this.state;

        return (

            <div>

                <input
                    type="text"
                    onChange=
                    {event=>this.handleChange(event)}
                    onKeyPress=
                    {event => this.onKeyPress(event)}
                    value={inputValue}
                    placeHolder={"Enter a video to search for"}/>

            </div>

        )
    }
}

Searchbar.propTypes = {

    API_KEY: PropTypes.string.isRequired,
    getVideos: PropTypes.func.isRequired
}
export default Searchbar;
