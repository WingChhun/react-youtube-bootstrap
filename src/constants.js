export const API_KEY = "AIzaSyC3ni21fos0ZbHdjOxCjXAEJBgyH4T1eo4";