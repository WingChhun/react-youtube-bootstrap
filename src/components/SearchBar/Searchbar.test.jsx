import React from 'react';
import ReactDOM from 'react-dom';

import {API_KEY} from "../../constants";

import Enzyme, {shallow, createMount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Seachbar from "./Searchbar";
import Searchbar from './Searchbar';

Enzyme.configure({adapter: new Adapter()});

describe("Searchbar", () => {

    const props = {
        API_KEY: API_KEY,
        getVideos: jest.fn()
    }

    it("Renders searchbar without crashing", () => {

        const wrapper = shallow(<Searchbar {...props}/>);

        expect(wrapper).toMatchSnapshot();

    })

    it("handleChange updates state", () => {
        const wrapper = shallow(<Searchbar {...props}/>);
        const instance = wrapper.instance();

        const event = {
            target: {
                value: "New"
            }
        }
        instance.handleChange(event);

        expect(instance.state.inputValue).toBe(event.target.value)

    })

    it("onKeyPress() calls function", () => {

        const wrapper = shallow(<Searchbar {...props}/>)
        const instance = wrapper.instance();

        const event = {
            key: "Enter"
        };

        instance.state.inputValue = "new";
        instance.onKeyPress(event)

        //expect(props.getVideos).toHaveBeenCalled()

    });

    it("KeyPresses and change will call functions ", () => {

        const wrapper = shallow(<Searchbar {...props}/>);
        const instance = wrapper.instance();
        const event = {
            key: "Enter"
        };

        const searchbar = wrapper.find(".searchBar")
        const onKeyPressSpy = jest.spyOn(instance, 'onKeyPress');

        instance.state.inputValue = "Hello";

        searchbar.simulate("keypress", {key: "hello"});

        expect(onKeyPressSpy).toBeCalled();
    });

    it("handleChange is called", () => {

        const wrapper = shallow(<Searchbar {...props}/>);
        const instance = wrapper.instance();
        const event = {
            target: {
                value: "New"
            }
        }

        const handleChangeSpy = jest.spyOn(instance, 'handleChange');
        const searchbar = wrapper.find('.searchBar');

        searchbar.simulate("change", {
            target: {
                value: "Hello"
            }
        });

        expect(handleChangeSpy).toHaveBeenCalled();

    })
});