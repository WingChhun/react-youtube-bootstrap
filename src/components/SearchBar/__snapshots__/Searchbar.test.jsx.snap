// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Searchbar Renders searchbar without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Searchbar
    API_KEY="AIzaSyC3ni21fos0ZbHdjOxCjXAEJBgyH4T1eo4"
    getVideos={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "searchBar",
      "onChange": [Function],
      "onKeyPress": [Function],
      "placeHolder": "Enter a video to search for",
      "type": "text",
      "value": "",
    },
    "ref": null,
    "rendered": null,
    "type": "input",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "searchBar",
        "onChange": [Function],
        "onKeyPress": [Function],
        "placeHolder": "Enter a video to search for",
        "type": "text",
        "value": "",
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
